version: "3.9"

services:
  db:
    image: mysql:8.0
    networks:
      - network_bimedoc
    ports:
      - 3306:3306
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
    volumes:
      - ./data/db:/var/lib/mysql
      - ./data/scripts:/docker-entrypoint-initdb.d
    healthcheck:
      test: [ "CMD", "mysqladmin", "ping", "-h", "localhost" ]
      interval: 3s
      timeout: 3s
      retries: 10

  web:
    build: .
    networks:
      - network_bimedoc
    ports:
      - 8000:8000
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
    volumes:
      - .:/code
    links:
      - db:db
    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    depends_on:
      db:
        condition: service_healthy

networks:
  network_bimedoc:
    driver: bridge
